### PR TITLE
PlaybackSettings: persist changes like playbackspeed, repeat, shuffle…

### PR DESCRIPTION
**Pull request by carolanitz**
… and aspectratio between playback

This also removes the user setting for the default playbackspeed since this is not needed anymore

### Checklist
- [x] I've run `bundle exec fastlane test` from the root directory to see all new and existing tests pass
- [x] I've followed the [vlc-ios code style](Docs/CodingStyle.md) and run `bundle exec fastlane lint` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/videolan/vlc-ios#contribute)
- [x] I've updated the documentation if necessary.

### CODE REVIEW

File: Apple-TV/PlaybackSettings.swift [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+ *
+ * Authors: Carola Nitz <caro # videolan.org>
+ *
+ * Refer to the COPYING file of the official project for license.
+ *****************************************************************************/
+
+import Foundation
+
+@objcMembers
+class PlaybackSettings: NSObject, Codable {
```
Comment (Mikanbu) [inline Apple-TV/PlaybackSettings.swift]: In this case since it's a data structure, what do you think about using a structure?

Comment (Mikanbu) [inline Apple-TV/PlaybackSettings.swift]: Would you mind explaining the choice of making this Codable with the choice of JSON representation ?

File: Apple-TV/PlaybackSettings.swift [hunk 2]
```diff
@@ -0,0 +1,73 @@
+/*****************************************************************************
```
Comment (Mikanbu) [inline Apple-TV/PlaybackSettings.swift]: I think this file shouldn't be inside Apple-TV folder.
Perhaps it should be inside SharedSources?

File: Sources/VLCPlaybackController.m [hunk 3]
```diff
@@ -793,12 +784,17 @@ - (void)pause
     [[NSNotificationCenter defaultCenter] postNotificationName:VLCPlaybackControllerPlaybackDidPause object:self];
 }
 
+- (void)setShuffleMode:(BOOL)shuffleMode
+{
+    _shuffleMode = _pbcSettings.shuffleMode = shuffleMode;
```
Comment (Mikanbu) [inline Sources/VLCPlaybackController.m]: Do we still need to keep a copy the different playback setting properties? Could we expose `PlaybackSettings` directly?

File: Apple-TV/PlaybackSettings.swift (hunk 1 above)
Comment (carolanitz) [reply Apple-TV/PlaybackSettings.swift]: yes that way I only need to store the playbacksettings as one object and not every single property

File: Apple-TV/PlaybackSettings.swift (hunk 2 above)
Comment (carolanitz) [reply Apple-TV/PlaybackSettings.swift]: true, But I think it is about time that we add swift to AppleTV

File: Sources/VLCPlaybackController.m (hunk 3 above)
Comment (carolanitz) [reply Sources/VLCPlaybackController.m]: Ah you mean we should remove those settings entirely from the playbackcontroller ? yes I like that idea

Comment (carolanitz) [reply Sources/VLCPlaybackController.m]: You need to set this information on the _listplayer or _mediaplayer. The settings are just holding that information. I would need to have a settingsDidChange method.

**Comment by carolanitz** [comment]
I will remove the aspectRatio though because it is media specific

**Comment by carolanitz** [comment]
This shouldn't be merged until we uniformly agree on the handling on states across all players

**Comment by carolanitz** [comment]
what we agreed on has been stated here: https://wiki.videolan.org/Playback_States/

**Pull request closed without merging**